bin/correct_n.py

```diff
@@ -0,0 +1,395 @@
+            {6031: 6, 12033: 18}
+    '''
+
+    # List of sequences as Biopython Seq objects (needed!) to be used for mafft input
+    sequence_list = [SeqIO.read(ref_in, 'fasta'), SeqIO.read(con_in, 'fasta')]
+
+    # Use tempfile as we don't want to keep the combined fasta file
+    fd, concat_temp = tempfile.mkstemp()
+
+    try:
+        with open(concat_temp, 'w') as handle:
+            SeqIO.write(sequence_list, handle, 'fasta')
+
+        # Run Mafft, adjustdirection sets it to map to reference sequence specifically
+        mafft_cline=MafftCommandline(input=concat_temp)
+        mafft_cline.adjustdirection = True
+
+        # MAFFT output goes to stdout
+        stdout, stderr = mafft_cline()
+    
+    # Remove temp file as MAFFT output is in stdout and we no longer need the file
+    finally:
+        os.remove(concat_temp)
+    
+
+    alignment = AlignIO.read(io.StringIO(stdout), 'fasta')
+
+    # Find the deletions in the alignment
+    # Start enumerate with one to get Genomic positions and not the python index
+    deletions = [index for index, character in enumerate(str(alignment[1].seq), start=1) if character == '-']
+    
+    # Find the non-consecutive numbers in the deletions list which are the gaps between the deletions
+    gaps = [[start, end] for start, end in zip(deletions, deletions[1:]) if start+1 < end]
+
+    # Using the gaps, create list iterator object of the start and end of consecutive ranges
+    edges = iter(deletions[:1] + sum(gaps, []) + deletions[-1:])
+
+    # Generate list of the start and end (use +1 to get the full range)
+    indel_groups = [[start, end+1] for start, end in zip(edges, edges)]
+    print('Deletion ranges include: {}'.format(indel_groups))
+
+    for group in indel_groups:
+        # Generate a dictionary containing the deletion start sites as keys and the deletion length as values 
+        # {5738: 9}
+        del_dict[group[0]] = group[1] - group[0]
+
+    return del_dict
+
+
+def _find_N(sequence, start=True):
+    '''
+    Find the First (start=True) non-N position or the last non-N (start=False) position
+
+    INPUTS:
+        sequence    --> `string` uppercase consensus nucleotide sequence
+        start       --> `boolean` switch to control if looking at the start of the sequence
+                        (true) or the end of the sequence (false)
+    
+    RETURNS:
+        `integer` position of the first non-N
+    '''
+
+    if start:
+        # Nucleotide location of first non-N is equal to number of N's at the start
+        for position, char in enumerate(sequence):
+            if char != 'N':
+                return position
+    else:
+        # Iteration from the end of the sequence, position of first N is equal to the length
+        # of the sequence minus the first location that is not an N when iterating from the end
+        for position, char in enumerate(sequence[::-1]):
+            if char != 'N':
```

Comment:
perhaps ensure that the char variable is upper case. Just in case one day we get 'n'